spec.md

```diff
@@ -843,6 +853,13 @@ message CreateVolumeRequest {
   // VOLUME_ACCESSIBILITY_CONSTRAINTS plugin capability, the SP MAY
   // choose where the provisioned volume is accessible from.
   TopologyRequirement accessibility_requirements = 7;
+
+  // Plugin specific mutable parameters passed in as opaque key-value
+  // pairs. This field is OPTIONAL. The Plugin is responsible for 
+  // parsing and validating these parameters. COs will treat these 
+  // as opaque. Plugins SHOULD treat these as if they take precedence 
+  // over the parameters field.
+  map<string, string> mutable_parameters = 8 [(alpha_field) = true];
```

Comment:
Can there be a CSI driver which can't apply all `mutable_parameters` during volume creation but can only apply some of the other parameters via separate `ControllerModifyVolume` RPC call? 

In terms of integration with CO, this would mean specifying problematic `mutable_parameters` during volume creation could cause PV provisioning to fail. I am fine if we chose not to support this use case but I wonder if it is worth calling out,  so as something like csi-sanity tests could validate a driver for CSI spec compliance.

Reply:
I don't think so. It's legal to call Modify right after Create, so an SP could just implement Create with mutable params as a create without mutable params followed immediately by a Modify (internally).

Requiring the CO so supply the mutable params at create time gives the SP a chance to avoid any expensive modifications by creating the volume correctly from the start. In theory we could have not included the param, but including it allows optimizations that wouldn't otherwise be possible. I see absolutely no downside from the SP perspective.